addons/point_of_sale/static/src/js/models.js

```diff
@@ -104,9 +105,32 @@ exports.PosModel = Backbone.Model.extend({
         // We fetch the backend data on the server asynchronously. this is done only when the pos user interface is launched,
         // Any change on this data made on the server is thus not reflected on the point of sale until it is relaunched.
         // when all the data has loaded, we compute some stuff, and declare the Pos ready to be used.
-        this.ready = this.load_server_data().then(function(){
-            return self.after_load_server_data();
-        });
+        // While fetching data from the server, we allow the data to be cached in IndexedDB. We, however, only allow a
+        // time window for when the data are cached. We do this using the state variable `stopCaching`. Basically,
+        // when `stopCaching` is false, POST requests are continuously cached in IDB. And once data loading is done,
+        // we set `stopCaching` to true which stops the POST request caching in the serviceWorker. See `cacheTheRequest`
+        // in pos-service-worker.js for more info.
+        this.ready = new Promise(async (resolve) => {
+            const swRegistration =
+                ('serviceWorker' in navigator && (await navigator.serviceWorker.getRegistration('/pos/'))) || false;
+
+            if (!(swRegistration && swRegistration.active && swRegistration.active.state === 'activated')) {
+                await this.load_server_data();
+            } else {
+                // We keep track of the pos_session_id.
+                // If the newly opened pos.session differs from the previously used pos.session,
+                // we invalidate the cache. This makes sure that the indexeddb cache is regularly cleared.
+                if (await idbget('pos_session_id') !== odoo.pos_session_id) {
```

Comment:
Why do we only clear when pos_session_id is different? (and not always). Because if we've added new products and still in the same pos session the cache won't be updated?

Reply:
I thought of clearing the cache always but then if I do, when the network is down, reloading will fail because there will be no data in the cache. Or do you think I can put the cache clearing somewhere else?